src/bp-members/bp-members-functions.php

```diff
@@ -1722,25 +1721,32 @@ function bp_core_validate_user_signup( $user_name, $user_email ) {
 
 		// Minimum of 4 characters.
 		if ( strlen( $user_name ) < 4 ) {
-			$errors->add( 'user_name',  __( 'Username must be at least 4 characters', 'buddypress' ) );
+			$errors->add( 'user_name', __( 'Username must be at least 4 characters.', 'buddypress' ) );
 		}
 
 		// No underscores. @todo Why not?
-		if ( false !== strpos( ' ' . $user_name, '_' ) ) {
+		if ( str_contains( ' ' . $user_name, '_' ) ) {
 			$errors->add( 'user_name', __( 'Sorry, usernames may not contain the character "_"!', 'buddypress' ) );
 		}
 
 		// No usernames that are all numeric. @todo Why?
 		$match = array();
 		preg_match( '/[0-9]*/', $user_name, $match );
-		if ( $match[0] == $user_name ) {
+
+		// Check for valid letters.
+		$letter_match = array();
+		preg_match( '/[a-zA-Z]+/', $user_name, $letter_match );
+
+		if ( $match[0] == $user_name || empty( $letter_match[0] ) ) {
```

Comment:
The current logic implies it is checking for the presense of letters inside the username, but it is only checking if the whole username is composed of numeric only.